### ISSUE TITLE
Avoid the use of InternalsVisibleTo

### ISSUE THREAD

**Issue by Corniel** [dotnet-project-file-analyzers/dotnet-project-file-analyzers#937]
It is a bad practice, anti-pattern, and should be avoided.

``` XML
 <ItemGroup>
    <!-- Expose internals to a single specific test project -->
    <InternalsVisibleTo Include="MyProject.UnitTests" />
    
    <!-- Dynamically expose internals using MSBuild properties -->
    <InternalsVisibleTo Include="$(AssemblyName).IntegrationTests" />
  </ItemGroup>
```

**Comment by Corniel** [comment]
If this is considered too opinionated, it can be reduced to only allow it for projects that are considered test projects.

**Comment by Sjaaky** [comment]
I prefer to allow it for test project.
But we could create 2 rules with different strictness.
